### PR TITLE
Remove setup panel shutdown listeners.

### PR DESCRIPTION
**Pull request by DanVanAtta**
## Overview
On shutdown all objects which we 'close' are falling out of scope and getting cleaned up anyways. We should get the same cleanup just through normal GC and the JVM terminating.

## Functional Changes
None

## Manual Testing Performed
- Check starts screens that game quits, can launch and quit cleanly.

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/framework/startup/mc/ServerModel.java [hunk 1]
```diff
@@ -132,16 +132,6 @@ public ServerModel(final GameSelectorModel gameSelectorModel, final SetupPanelMo
     headless = (interactionMode == InteractionMode.HEADLESS);
   }
 
-  public void shutDown() {
-    gameSelectorModel.deleteObserver(gameSelectorObserver);
-    if (serverMessenger != null) {
-      chatController.deactivate();
```
[the remark this answers is not in the document]
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/framework/startup/mc/ServerModel.java]: Okay, I hope to have resolved this with: 86eaee51584a17e6b5ee7dba70da4f0b0cb7f454

- #cancel method does exact same thing as #shutdown, so I added a call back in to invoke shutdown.

It looks like this was the *one* place where we actually did something that was more than changing in-memory references.  It was very good to double check this thoroughly.

**Comment by DanVanAtta** [comment]
One thing to clarify, this is a `shutdown` method, we call `System.exit(0)` after `shutdown`. There is no relying on the garbage collector, the JVM is exit'ing.

If you go through this code carefully, it's all mostly an elaborate shell game to null out references that will not matter. Checking it carefully I noted one place where we did do network calls to notify shutdown. We have a copy/pasted cancel method that looks to do the real cleanup we are thinking of that could cause connection leaks if not necessarily cleaned up. So, to fix that one location we will call #cancel instead  : )

One thing to note, the headless implementations of shutdown are largely no-op. So it turns out an unexpected large number of paths are no-ops. The other majority case is removing memory references that are going to be cleaned anyways as the JVM is shutting down. But, we did have the one place where there was some useful cleanup work, that has been fixed/up restored.

**Comment by DanVanAtta** [comment]
@RoiEXLab those are valid concerns. It's interesting checking this code in detail that most of it looks like it should do something more important, but it's not the case : )  Hence the cleanup/removal, largely we have a lot of dead code. Validating that concern proved useful, I certainly had a mistake to remove one too many shutdown calls. At this point we should have all in-memory reference removed, which is just as well since `System.exit` will clean those up.

**Comment by RoiEXLab** [comment]
@DanVanAtta I'm under the impression that "shutdown" calls are called whenever we clear an instance of something.
For example closing a game and getting to the main menu is triggering (or at least was, I recall fiddling with code in ClientSetupPanel that gets called when quitting a game) a shutdown chain is being invoked, without any complete exit, so that could cause problems.
I don't know if this specific code is related to some occurrences I saw a couple weeks ago but this is what came to my mind when I saw this PR

**Comment by DanVanAtta** [comment]
Gotchya, it's probably worth double checking what I saw, but this path seems to be a duplicate of 'cancel' and invoked before we terminate the JVM via `System.exit`. It turns out a lot of this just turns out to be noise, removing memory references is not needed if we `System.exit`. I was working through this code and found this amongst a lot of other code can just be carefully deleted.

**Comment by DanVanAtta** [comment]
So specifically, we trigger the `shutdown` calls

**Pull request merged**